src/transformers/tokenization_t5.py

```diff
@@ -206,3 +210,90 @@ def save_vocabulary(self, save_directory):
+
+        Args:
+            token_ids_0 (:obj:`List[int]`):
+                List of IDs to which the special tokens will be added
+            token_ids_1 (:obj:`List[int]`, `optional`, defaults to :obj:`None`):
+                Optional second list of IDs for sequence pairs.
+
+        Returns:
+            :obj:`List[int]`: list of `input IDs <../glossary.html#input-ids>`__ with the appropriate special tokens.
+        """
+        if token_ids_1 is None:
+            return self.prefix_tokens + token_ids_0 + self.suffix_tokens
+        # We don't expect to process pairs, but leave the pair logic for API consistency
+        return self.prefix_tokens + token_ids_0 + token_ids_1 + self.suffix_tokens
+
+    def prepare_seq2seq_batch(
+        self,
+        src_texts: List[str],
+        tgt_texts: Optional[List[str]] = None,
+        max_length: Optional[int] = None,
+        max_target_length: Optional[int] = None,
+        padding: str = "longest",
+        return_tensors: str = None,
+        **kwargs,
+    ) -> BatchEncoding:
+        """Prepare a batch that can be passed directly to an instance of T5Model.
+        Arguments:
+            src_texts: list of src language texts
+            tgt_texts: list of tgt language texts
+            max_length: (default=None, which defers to the config value of 512 for t5*
+            padding: strategy for padding input_ids and decoder_input_ids. Should be max_length or longest.
+            **kwargs: passed to self.__call__
+
+        Returns:
+            :obj:`BatchEncoding`: with keys input_ids, attention_mask, decoder_input_ids, decoder_attention_mask.
+        """
+        if max_length is None:
+            max_length = self.max_len
+        self.set_src_special_tokens()
+        model_inputs: BatchEncoding = self(
+            src_texts,
+            add_special_tokens=True,
+            return_tensors=return_tensors,
+            max_length=max_length,
+            padding=padding,
+            truncation=True,
+            **kwargs,
+        )
+        if tgt_texts is None:
+            return model_inputs
+        # Process tgt_texts
+        if max_target_length is None:
+            max_target_length = max_length
+        self.set_tgt_special_tokens()
+        decoder_inputs: BatchEncoding = self(
+            tgt_texts,
+            add_special_tokens=True,
+            return_tensors=return_tensors,
+            padding=padding,
+            max_length=max_target_length,
+            truncation=True,
+            **kwargs,
+        )
+        for k, v in decoder_inputs.items():
+            model_inputs[f"decoder_{k}"] = v
+
+        self.set_src_special_tokens()
+        return model_inputs
+
+    def set_src_special_tokens(self) -> None:
+        self.prefix_tokens = []
+        self.suffix_tokens = [self.eos_token_id]
+
+    def set_tgt_special_tokens(self) -> None:
+        self.prefix_tokens = [self.pad_token_id]
+        self.suffix_tokens = [self.eos_token_id]
```

Comment:
not entirely sure about adding `eos` automatically. What do you think @sshleifer ?

Reply:
1) I wouldn't do `eos` in this PR. I think for that we need to either
a) get to the bottom of why it impacts zero shot translation performance 
or 
b) add a flag to support not adding it (for backward compatibility/ zero shot tasks).


2) Do we have evidence that adding a prefix token on the decoder side is helpful?

Reply:
> Do we have evidence that adding a prefix token on the decoder side is helpful?

yes, the `T5Model` does this in the `_shift_right` method. Same is the case with the original TF T5 implementation. AFAIK in seq2seq models decoder uses special start token, in BART the tokenizer automatically adds `bos`, in T5 there is no `bos` instead `pad` token is used as decoder start id

Reply:
get to the bottom of why it impacts zero shot translation performance

I will remove it for now, and for this issue to be solved.